src/plugins/warp-plugin.ts

```diff
@@ -0,0 +1,180 @@
+    const minecraftInstances = context.localInstances.filter(
+      (instance) => instance instanceof MinecraftInstance
+    ) as MinecraftInstance[]
+
+    if (context.addChatCommand) context.addChatCommand(new WarpCommand(minecraftInstances))
+
+    context.application.on('instance', (event) => {
+      if (event.type === InstanceEventType.create && event.instanceType === InstanceType.MINECRAFT) {
+        const localInstance = context.localInstances.find(
+          (instance) => instance instanceof MinecraftInstance && instance.instanceName === event.instanceName
+        )
+        if (localInstance != undefined) {
+          const clientInstance = localInstance as MinecraftInstance
+          // "login" packet is also first spawn packet containing world metadata
+          clientInstance.client?.on('login', async () => {
+            if (!disableLimboTrapping) await limbo(clientInstance)
+          })
+          clientInstance.client?.on('respawn', async () => {
+            if (!disableLimboTrapping) await limbo(clientInstance)
+          })
+        }
+      }
+    })
+  }
+} satisfies PluginInterface
+
+async function limbo(clientInstance: MinecraftInstance): Promise<void> {
+  clientInstance.logger.debug('Spawn event triggered. sending to limbo...')
+  await clientInstance.send('§')
+}
+
+async function warpPlayer(app: Application, minecraftInstanceName: string, username: string): Promise<string> {
+  disableLimboTrapping = true
+
+  // exit limbo and go to main lobby. Can't warp from limbo
+  app.clusterHelper.sendCommandToMinecraft(minecraftInstanceName, '/lobby')
+
+  // Go to Skyblock first before warping.
+  // Person can rejoin if warped to the main lobby
+  await sleep(2000)
+  app.clusterHelper.sendCommandToMinecraft(minecraftInstanceName, '/skyblock')
+
+  // ensure the account is in the hub and not on private island
+  // to prevent being banned for "profile boosting"
+  await sleep(12_000) // need higher cooldown to change between lobbies
+  app.clusterHelper.sendCommandToMinecraft(minecraftInstanceName, '/hub')
+
+  await sleep(2000)
+
+  const errorMessage = await awaitPartyStatus(app, minecraftInstanceName, username)
+  if (errorMessage != undefined) {
+    disableLimboTrapping = false
+    app.clusterHelper.sendCommandToMinecraft(minecraftInstanceName, '§')
+
+    app.clusterHelper.sendCommandToMinecraft(minecraftInstanceName, '/party disband')
+    app.clusterHelper.sendCommandToMinecraft(minecraftInstanceName, '/party leave')
+
+    return errorMessage
+  }
+
+  app.clusterHelper.sendCommandToMinecraft(minecraftInstanceName, '/party warp')
+  app.clusterHelper.sendCommandToMinecraft(minecraftInstanceName, `/pc Blame the gods on your luck`)
+  app.clusterHelper.sendCommandToMinecraft(minecraftInstanceName, '/party disband')
+  app.clusterHelper.sendCommandToMinecraft(minecraftInstanceName, '/party leave')
+
+  disableLimboTrapping = false
+  app.clusterHelper.sendCommandToMinecraft(minecraftInstanceName, '§')

```

Comment:
Move this directly below the following code:
``` typescript
app.clusterHelper.sendCommandToMinecraft(minecraftInstanceName, '/party warp')
```
This removes unecessary delay between re-entering limbo.

Reply:
While switching lobbies, sending commands will be blocked. It is rather better to use 2 seconds to send the commands before switching lobbies than to switch lobby first and wait 10 seconds for the cooldown on commands.